selectors/rhs.jsx

```diff
@@ -46,4 +47,15 @@ export const getSelectedPost = createSelector(
 export function makeGetCommentDraft(rootId) {
     const defaultValue = {message: '', fileInfos: [], uploadsInProgress: []};
     return makeGetGlobalItem(`comment_draft_${rootId}`, defaultValue);
-}
\ No newline at end of file
+}
+
+export const getPostsEmbedVisibleObj = createSelector(
+    (state) => state,
+    (state, {posts, previewCollapsed}) => ({posts, previewCollapsed}),
+    (state, {posts, previewCollapsed}) => {
+        return posts.reduce((accumulatedEmbedVisibleFlags, post) => ({
+            ...accumulatedEmbedVisibleFlags,
+            [post.id]: makeGetItem(StoragePrefixes.EMBED_VISIBLE + post.id, previewCollapsed.startsWith('false'))(state)
```

Comment:
@hmhealey I have to send entire state and therefore computes with every state change. Should we have `makeGetItem ` as memorised selector which depends on `state.store` instead?.

Reply:
makeGetItem doesn't work with selectors with how it's currently designed, so I don't think we can use it here, but we can still definitely have this memoized properly.

What we can do though is change what the selector takes from the store so that it only takes the things that affect it: the storage state, the current user ID for the storage key prefix, and the COLLAPSE_DISPLAY preference. We'll also want to have a function that creates a selector here because it could possibly be used in multiple places, and we want to make sure each one is memoized separately so they don't clear each other's stored results. It could look something like

```javascript
export function makeGetPostsEmbedVisibleObj() {
    return createSelector(
        // This could be its own function or a selector, but that's not necessary since it's really simple
        (state) => state.storage,
        // We don't need to pass in the previewCollapsed value here since we can just get it ourselves.
        // Also, we can use getBool so that we don't need to work with a string later on
        (state) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLLAPSE_DISPLAY),
        (state, posts) => posts,
        (storage, previewCollapsed, posts) => {
            const postsEmbedVisibleObj = {};

            for (const post of posts) {
                // You'll need to write something like makeGetItem that doesn't require the whole store to be passed in
                postsEmbedVisibleObj[post.id] = getItemFromStorage(storage, currentUserId, StoragePrefixes.EMBED_VISIBLE + post.id, previewCollapsed);
            }

            return postsEmbedVisibleObj;
        }
    );
}
```

That way, it'll only change when one of those exact 3 things changes, and not just any state change.

Reply:
I will make the changes. Thanks, this looks great 👍 .  Just one thing though I wanted to change this to ` (state) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLLAPSE_DISPLAY),` bool. As its being send to couple of nested components i was sceptical about the changes to be made. I will give it a shot though.